### ISSUE TITLE
unused CSS classes? dead code?

### ISSUE THREAD

**Issue by prather-mcs** [jasmine/jasmine#847]
I'm currently writing a fix for Issue 844 :
https://github.com/jasmine/jasmine/issues/844

In this process, I think I've found some CSS classes that are essentially "dead code" because I can't find any JavaScript code in the codebase that ever applies them. I'm referring to lines 258&ndash;288 here: 
https://github.com/jasmine/jasmine/blob/426eebe1ce2d202c4c36ff123974ec88b073ea62/src/html/_HTMLReporter.scss#L258

The SCSS code there compiles to dead code in `jasmine.css`, lines 43&ndash;48:
https://github.com/jasmine/jasmine/blob/426eebe1ce2d202c4c36ff123974ec88b073ea62/lib/jasmine-core/jasmine.css#L43

Basically, I can't find any JavaScript in the whole code base that ever refers to CSS classes `showDetails` or `summaryMenuItem`. I know that code can still refer to things implicitly, like when you invoke `forEach` or invoke a `for..in` loop, however these class names are located in CSS, which as far as I know doesn't really have any programmatic constructs for looping or otherwise accessing implicitly &mdash; we always have to explicitly name the CSS class in JS to apply them, right?

*(You could also get a class name in JS by querying a DOM element and then accessing its `classList` property and then accessing the returned array-like object, and in that case you would also have gotten a CSS class in your JavaScript without ever naming it explicitly &mdash; but the class would have to be explicitly set in the HTML, and that's not applicable in this case, because Jasmine constructs all of its HTML programmatically.)*

And I can't find DOM elements with these class names (`.showDetails`, `.summaryMenuItem`) ever appearing in the browser document .

So I'm really thinking that these style rules are dead wood left over from a time when they were used, but are not used any more.

---

My questions are:

* have I said something wrong above?
* can I just take out this dead CSS in the other fix I'm working on, then submit a pull request to fix both 844 and this Issue at once?
* or should I just make the most atomic of commits, with one that removes the dead CSS, and then one that fixes 844 (my fix in 844 is to make "unique enough" names for the CSS that Jasmine uses)

**edit/update**
I'll take them out while I'm developing, and see if all specs still pass. I probably should have done that before opening Issue, in addition to the code reasoning I did.

**Comment by slackersoft** [comment]
In general we prefer smaller commits/pull requests. If you have some classes that are no longer used, I'd rather see a PR for just their removal separate from the PR you're working on to rename the classes that are in use.

**Comment by prather-mcs** [comment]
Hi I just finished my fix for #844 .

And I caught one more (one last) dead CSS style rule .

The `running-alert` class seems not to be used anywhere else in the codebase, and by my reasoning no Jasmine code will ever produce html with that class, and so we can take it out.

So I'm going to create one more pull request that removes this dead style rule, before I introduce the much bigger code change I've already got which is a fix for 844. I'm going to invite people with more experience developing Jasmine to take a good hard look at my fix for 844, and make sure it's really what the project wants, of course. Maybe there's some alternative way to fix 844, and maybe I screwed something up in my fix for it.

But for now the small pull request I'm about to introduce (removing `.running-alert` style rule) should be good-to-go and should officially pave the way clear for my 844 fix.
